src/commissioner/commissioner_argcargv.cpp

```diff
@@ -297,244 +74,207 @@ static void handle_pskd(ArgcArgv *pThis, CommissionerArgs *args)
+            break;
+        case 'H':
+            aArgs.mAgentHost = optarg;
+            break;
+        case 'P':
+            aArgs.mAgentPort = optarg;
+            break;
+        case 'L':
+            steeringLength = atoi(optarg);
+            VerifyOrExit(steeringLength >= 1 and steeringLength <= 16,
+                         fprintf(stderr, "Steering data length must be between 1 and 16!"));
+            break;
+        case 'q':
+            otbrLogEnableSyslog(false);
+            break;
+        case 'l':
+            otbrLogSetFilename(optarg);
+            break;
+        case 'd':
+            aArgs.mDebugLevel = atoi(optarg);
+            VerifyOrExit(aArgs.mDebugLevel >= 1 and aArgs.mDebugLevel <= 7,
+                         fprintf(stderr, "Debug level must be between 1 and 7!"));
+            break;
+        case 'i':
+            aArgs.mKeepAliveInterval = atoi(optarg);
+            VerifyOrExit(aArgs.mKeepAliveInterval >= 0, fprintf(stderr, "Invalid value for keep alive interval!"));
+            break;
+        case 'h':
+            PrintUsage(aArgv[0], stdout, EXIT_SUCCESS);
+            break;
+        case '?':
+            break;
+        default:
+            assert(false);
+            break;
+        }
+    }
 
-/** handle debug level on command line */
-static void handle_debug_level(ArgcArgv *pThis, CommissionerArgs *args)
-{
-    (void)args;
-    int n;
+    VerifyOrExit(networkName != NULL, fprintf(stderr, "Missing network name!"));
+    VerifyOrExit(networkPassword != NULL, fprintf(stderr, "Missing network password!"));
+    VerifyOrExit(isXPanIdSet, fprintf(stderr, "Missing extended PAN ID!"));
 
-    n = pThis->NumParam();
-    if (n < OTBR_LOG_EMERG)
+    if (steeringLength == 0)
     {
-        pThis->usage("invalid log level, must be >= %d\n", OTBR_LOG_EMERG);
+        steeringLength = (allowAllJoiners ? 1 : kSteeringDefaultLength);
     }
-    if (n > OTBR_LOG_DEBUG)
-    {
-        n = OTBR_LOG_DEBUG;
-    }
-    otbrLogSetLevel(n);
-}
 
-/** handle steering allow any on command line */
-static void handle_allow_all_joiners(ArgcArgv *pThis, CommissionerArgs *args)
-{
-    (void)(pThis);
-    args->mAllowAllJoiners = true;
-}
-
-/** user wants to disable COMM_KA transmissions for test purposes */
-static void handle_comm_ka_disabled(ArgcArgv *pThis, CommissionerArgs *args)
-{
-    (void)(pThis);
-    args->mNeedSendCommKA = false;
-}
+    aArgs.mSteeringData.Init(steeringLength);
 
-/** user wants to adjust COMM_KA transmission rate */
-static void handle_comm_ka_rate(ArgcArgv *pThis, CommissionerArgs *args)
-{
-    int n;
-
-    n = pThis->NumParam();
-    /* sanity... */
+    if (!allowAllJoiners)
+    {
+        VerifyOrExit(aArgs.mPSKd != NULL, fprintf(stderr, "Missing PSKd!"));
+        VerifyOrExit(isEui64Set, fprintf(stderr, "Missing EUI64!"));
 
-    /* note: 86400 = 1 day in seconds */
-    if ((n < 3) || (n > 86400))
+        aArgs.mSteeringData.ComputeJoinerId(joinerEui64, joinerEui64);
```

Comment:
Previous code puts steering data computation(and any other non-trivial task) outside argument parsing to make it easier to identify main procedures in commissioning. Is it a preferred way of organizing code?

Reply:
I think computing steering data is trivial and should be done when parsing arguments. We may support multiple joiners later, if we don't compute steering data as we parsing argument, we have to store these arguments, which is not clean.
Besides, some command line arguments are not used after parsing, we don't need to save them all the time.